File: Java/DynamicProgramming/BJAlgo_1904.java
import java.io.*;

public class BJAlgo_1904 {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int N = Integer.parseInt(br.readLine());
        int[] cache = new int[N + 1];
        cache[0] = 1;
        cache[1] = 1;

        if (N > 1) {
            for (int i = 2; i < N + 1; i++) {
                cache[i] = (cache[i - 1] + cache[i - 2]) % 15746;
            }
        }

        System.out.println(cache[N]);
    }
}

File: Python/DynamicProgramming/BJAlgo_1904.py
from sys import stdin

N = int(stdin.readline())

cache = [0 for index in range(N + 1)]
cache[0] = 1
cache[1] = 1

if N > 1:
    for index in range(2, N + 1):
        cache[index] = (cache[index - 1] + cache[index - 2]) % 15746

print(cache[N])